### DISCUSSION TITLE
Writing edge-centered quadmesh data and plotting in VisIt

### DISCUSSION THREAD

**Discussion by patrickb314** [LLNL/Silo#241] [Q&A]
I have edge-centered 2D quadmesh data I'm trying to write and VisIt is complaining about the size of the resulting data. As a concrete example, I'll describe what happens when I try to write 128 by 128 zone (129 node by 129 node) quadmesh with zone and edge variables values. The zone values are working well, but I'm having trouble with the (scalar) edge data. I'm packing the edge data into a 2 * 129 * 129 (33282 element) array, putting the i edges first in the first 129x129 space with phony 0 values in the non-existent locations, and then the j edges after that. In the call to Silo, ndims is 2 and dims is {129, 129} since this is true edge-centered data. I think all of that matches what the Silo documentation says I should be doing.

The problem is that when I try to read this data into VisIt as pseudocolor data, it complains that "The compute engine running on host patricks-macbook-pro.local issued the following warning: In domain 0, your nodal variable "domain_00000/velocity" has 33282 values, but it should have 16641.  Some values were removed to ensure VisIt runs smoothly."

Is this a problem with my understanding of how to call Silo to write edge-centered data, or is this an issue I should take up with VisIt?

**Comment by markcmiller86** [comment]
@patrickb314 apologies for troubles but I suspect the issue is that the VisIt plugin has not yet been enhanced to handle edge centered data from Silo.

Do you by chance have a small example you'd be willing to share here (via attachment) or, if exceeding size constraings maybe using one of the options [here](https://visit-dav.github.io/largedata/help/using-for-uploads.html).

**Reply by patrickb314** [reply]
Sure, I've attached a quick and dirty C file to recreate this.

[silotest.c.gz](https://github.com/LLNL/Silo/files/7953557/silotest.c.gz)

**Reply by patrickb314** [reply]
(For the record, it looks like this has the u and v velocities in the edge array in the wrong order since the u velocities are j-oriented edges, but that's beside the point for the error I'm seeing, I think).
